core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -150,232 +147,148 @@ impl RocksDBStore {
         db_opts.create_if_missing(true);
         db_opts.create_missing_column_families(true);
 
-        let transactional_db_opts = TransactionDBOptions::default();
-
         let column_families: Vec<ColumnFamilyDescriptor> = ALL_COLUMN_FAMILIES
             .into_iter()
             .map(|cf| ColumnFamilyDescriptor::new(cf.to_string(), Options::default()))
             .collect();
 
-        let db = TransactionDB::open_cf_descriptors(
-            &db_opts,
-            &transactional_db_opts,
-            root.as_path(),
-            column_families,
-        )
-        .unwrap();
+        let db = DB::open_cf_descriptors(&db_opts, root.as_path(), column_families).unwrap();
 
         RocksDBStore { db }
     }
 
-    fn cf_handle(&self, cf: &RocksDBColumnFamily) -> &ColumnFamily {
-        self.db.cf_handle(&cf.to_string()).unwrap()
-    }
-}
-
-impl<'db> CommitStore<'db> for RocksDBStore {
-    type DBTransaction = RocksDBCommitTransaction<'db>;
-
-    fn create_db_transaction(&'db mut self) -> RocksDBCommitTransaction<'db> {
-        let db_txn = self.db.transaction();
-        let column_families = ALL_COLUMN_FAMILIES
-            .iter()
-            .map(|cf| (cf.clone(), self.cf_handle(cf)))
-            .collect();
-        RocksDBCommitTransaction {
-            db_txn,
-            column_families,
-        }
-    }
-}
-
-pub struct RocksDBCommitTransaction<'db> {
-    db_txn: rocksdb::Transaction<'db, TransactionDB>,
-    column_families: BTreeMap<RocksDBColumnFamily, &'db ColumnFamily>,
-}
-
-impl<'db> RocksDBCommitTransaction<'db> {
-    fn insert_transaction(&mut self, transaction_bundle: CommittedTransactionBundle) {
+    fn add_transaction_to_write_batch(
+        &self,
+        batch: &mut WriteBatch,
+        transaction_bundle: CommittedTransactionBundle,
+    ) {
         let (transaction, receipt, identifiers) = transaction_bundle;
         let state_version = identifiers.state_version;
+        let ledger_payload_hash = transaction.ledger_payload_hash();
 
         // TEMPORARY until this is handled in the engine: we store both an intent lookup and the transaction itself
         if let LedgerTransaction::User(notarized_transaction) = &transaction {
-            let existing_intent_hash_mapping_opt = self
-                .db_txn
-                .get_for_update_cf(
-                    self.cf_handle(&StateVersionByTxnIntentHash),
-                    notarized_transaction.intent_hash(),
-                    true,
-                )
-                .expect("RocksDB: failure to read intent hash");
-            if let Some(existing_intent_hash_mapping) = existing_intent_hash_mapping_opt {
+            /* For user transactions we only need to check for duplicate intent hashes to know
+            that user payload hash and ledger payload hash are also unique. */
+            let intent_hash = notarized_transaction.intent_hash();
+
+            let maybe_existing_intent_hash = self
```

Comment:
This check has been watered down now and doesn't check for a collision of intent hashes inside the batch itself.

Maybe we can do this in the commit function, by storing a set of intent hashes as we go? And checking the size matches the batch size at the end?